x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -734,9 +738,10 @@ export const VisualizationWrapper = ({
     useReportingState(errors);
 
   const onRenderHandler = useCallback(() => {
+    setDynamicError(false);
     setIsRenderComplete(true);
     onRender$();
-  }, [setIsRenderComplete, onRender$]);
+  }, [onRender$, setDynamicError, setIsRenderComplete]);
```

Comment:
qq: Could be great to know what a DynamicError is since it looks like used only in this workspace_panel file.
Why is "dynamic"?

Reply:
I guess what we meant here is runtime and coming from the fact we cannot generate expression (so most likely it would be a wrong kql syntax or a user mistake. When it appears, `ExpressionRendererComponent` instead of trying to render the expression, renders the content of the `renderError` prop.